core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -901,6 +900,67 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleCreateTopicRequest(request: RequestChannel.Request) {
+    val createTopicRequest = request.body.asInstanceOf[CreateTopicRequest]
+
+    def handleUnauthorizedRequest = {
+      createTopicRequest.topics.asScala.map { case (topic, _) =>
+        (topic, Errors.CLUSTER_AUTHORIZATION_FAILED)
+      }
+    }
+
+    def handleAuthorizedRequest = {
+      val brokers = metadataCache.getAliveBrokers.map(_.id)
+      createTopicRequest.topics.asScala.map { case (topic, arguments) =>
+        try {
+          // Prep configs
+          val configs = new Properties()
+          arguments.configs.asScala.foreach { case (key, value) =>
+            configs.setProperty(key, value)
+          }
+          LogConfig.validate(configs)
+
+          // Prep assignment
+          val assignments = {
+            if (!arguments.replicasAssignments.isEmpty) {
+              // Note: we don't check that replicaAssignment doesn't contain unknown brokers - unlike in add-partitions case,
+              // this follows the existing logic in TopicCommand
+              arguments.replicasAssignments.asScala.map { case (partitionId, replicas) =>
+                (partitionId.intValue, replicas.asScala.map(_.intValue).toSeq)
+              }
+            } else {
+              AdminUtils.assignReplicasToBrokers(brokers, arguments.partitions, arguments.replicationFactor)
+            }
+          }
+
+          // Create the topic
+          AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(zkUtils, topic, assignments, configs, update = false)
+
+          // Return valid response
+          (topic, Errors.NONE)
+        } catch {
+          // Handle errors with appropriate error response
+          case e: Throwable => (topic, Errors.forException(e))
+        }
+      }
+    }
+
+    val results =
+      if(!authorize(request.session, Create, Resource.ClusterResource)) {
+        handleUnauthorizedRequest
+      } else {
+        handleAuthorizedRequest
+      }
+
+    val respHeader = new ResponseHeader(request.header.correlationId)
+    val responseBody = new CreateTopicResponse(results.toMap.asJava)
+
+    trace("Sending create topics response %s for correlation id %d to client %s."
```

Comment:
I've been encouraging people to use String interpolation for new code because I think it's less error-prone. Unless there's a good reason to use String.format. Is this one such case?